src/Image.zig

```diff
@@ -42,17 +44,49 @@ pub const Stream = io.StreamSource;
 
 pub const EncoderOptions = AllImageFormats.ImageEncoderOptions;
 
-pub const SaveInfo = struct {
-    width: usize,
-    height: usize,
-    encoder_options: EncoderOptions,
+pub const AnimationLoopInfinite = -1;
+
+pub const AnimationFrame = struct {
+    pixels: PixelStorage,
+    duration: f32,
+};
+
+pub const Animation = struct {
+    frames: std.ArrayListUnmanaged(AnimationFrame),
+    loop_count: i32 = AnimationLoopInfinite,
+
+    pub fn deinit(self: *Animation, allocator: std.mem.Allocator) void {
+        for (self.frames.items) |frame| {
+            frame.pixels.deinit(allocator);
+        }
+
+        self.frames.deinit(allocator);
+    }
+};
+
+pub const Data = union(enum) {
+    empty: void,
+    image: PixelStorage,
```

Comment:
What do you think we leave the pixels in the Image and just add `animation: ?Animation` field? Animation could have a copy of the PixelStorage from pixels field (since its size is just tag + 2 x slice, which I think is 17bytes) as its first frame but it will allow code that isn't interested in animation to threat all images the same. It also seems that it would simplify code in this file significantly.

Reply:
Also it just crossed my mind that instead of `pixels: ?PixelStorage = null` we could have `pixels: PixelStorage = .{.rgba32 = &[_]u8{}}`. That way we can avoid forcing users to do null checks.